packages/system/src/generator.ts

```diff
@@ -21,7 +21,21 @@ export class StyleGenerator {
     const pseudoProps: { [key: string]: any } = {};
 
     for (const [propName, propValue] of Object.entries(props)) {
-      if (isStyledProp(propName.trim())) {
+      if (
+        typeof propValue === "string" &&
+        /[a-zA-Z]+\.[a-zA-Z0-9]+/.test(propValue)
+      ) {
+        const userTheme = theme.getUserTheme();
+        const propKey = propValue.split(".")[0] as Tokens;
+        if (userTheme[propKey] !== undefined) {
+          for (const key in userTheme[propKey]) {
+            if (propValue.trim() === key) {
+              styledProps[propName.trim()] = userTheme[propKey]![key];
+              break;
+            }
+          }
```

Comment:
The reason for needing `trim()` on propValue is actually because `getFullText` is used instead of `getText` within the `collector` (My fault 🤦). I believe that fixing those area might eliminate the need for this adjustment. 
Whether or not this will be addressed in this PR is up to @nojiritakeshi  though

Reply:
@yukukotani @poteboy 
I'll try to fix it and include it in this PR, but I'll separate the commits.

Reply:
@poteboy 
I removed all the trim()'s that I thought were unnecessary, but all the existing tests were fine.
I also checked the actual operation of the system and it was fine as far as I was concerned!

Reply:
Thank you 🍑